### PR TITLE
Themes: Show site selector for /upload url direct hit

### PR DESCRIPTION
**Pull request by seear**
Logged-in visits to wordress.com/themes/upload were redirecting to /theme/upload and showing an 'empty' message.

Tweak the server routing so that this route is not redirected, and tweak client routing so that logged-out hits redirect to /themes.

**Before**
screen shot 2017-06-02 at 14 22 54

**After**
screen shot 2017-06-02 at 14 23 09

**To Test**
* Check /themes/upload logged-in and logged-out
* Also check /themes/upload/{site} logged-in and logged-out
* All other themes routes should work as before

### CODE REVIEW

**Comment by folletto** [comment]
Tested, works. 👍

**Comment by seear** [comment]
pinging @ockham for a quick routing review...

**Comment by ockham** [comment]
Sorry, missed this for a while. Will review today.

File: client/my-sites/themes/index.node.js [hunk 1]
```diff
@@ -35,7 +35,7 @@ export default function( router ) {
 			loggedOut,
 			makeLayout
 		);
-		router( '/themes/upload/*', makeLayout );
+		router( '/themes/upload*', makeLayout );
```
Comment (ockham) [inline client/my-sites/themes/index.node.js]: I like to keep them explicit (to disallow `/themes/uploadfoobar` -- something like that bit me recently with a v4 legacy route :roll_eyes:):

```js
router( [ '/themes/upload', '/themes/upload/*' ], makeLayout );
```

File: client/my-sites/themes/index.web.js [hunk 2]
```diff
@@ -41,6 +41,8 @@ export default function( router ) {
 				makeLayout
 			);
 		} else {
+			router( '/themes/upload*', '/themes' );
```
Comment (ockham) [inline client/my-sites/themes/index.web.js]: Would be nice if the redirect was a bit more explicit, but I see that relying on `router.redirect()` to be translated to `page.redirect()` is risky, too (implementation detail). Let's keep it and maybe add a little comment (upload not possible when logged-out, hence redirect).

**Comment by seear** [comment]
Thanks for review @ockham <3

**Pull request merged**